src/server/game/Entities/Object/Object.cpp

```diff
@@ -2539,6 +2612,69 @@ float WorldObject::GetFloorZ() const
     return std::max<float>(m_staticFloorZ, GetMap()->GetGameObjectFloor(GetPhaseMask(), GetPositionX(), GetPositionY(), GetPositionZ()));
 }
 
+void WorldObject::SetTransport(Transport* transport)
+{ 
+    if (transport) 
+    {
+        m_movementInfo.transport.guid = transport->GetGUID();
+        // could it be possible to initialize the rest of the transport data (seat, time, offset) here?
+        AddUnitMovementFlag(MOVEMENTFLAG_ONTRANSPORT);
+    }
+    else // if(!m_vehicle) ?
+    {
+        m_movementInfo.transport.Reset();
+        RemoveUnitMovementFlag(MOVEMENTFLAG_ONTRANSPORT);
+    }
+
+    m_transport = transport;
+}
+
+MovementInfo WorldObject::GetMovementInfo() const
```

Comment:
The goal of the method is to return the movement state of the unit at the moment of the call. It doesn't have to be a reference. Also, returning a different object allows us to play with the timestamp inside (which may come handy in the future). the caller can decide which value will be in the timestamp.